app/plugin_key_value_store.go

```diff
@@ -23,14 +23,13 @@ func (a *App) SetPluginKey(pluginId string, key string, value []byte) *model.App
 }
 
 func (a *App) SetPluginKeyWithExpiry(pluginId string, key string, value []byte, expireInSeconds int64) *model.AppError {
-
 	if expireInSeconds > 0 {
 		expireInSeconds = model.GetMillis() + (expireInSeconds * 1000)
 	}
 
 	kv := &model.PluginKeyValue{
 		PluginId: pluginId,
-		Key:      getKeyHash(key),
+		Key:      key,
```

Comment:
If we simply replace the set method using `key` instead of `getKeyHash(key)` we will end up with potentially duplicates and/or inconsistent data (from a plugin PoV) - We should use that function call to ensure that the hash version will be deleted (via `result := <-a.Srv.Store.Plugin().Delete(pluginId, getKeyHash(key))`) - That way we can slowly clean up the old K/V entries from the database until removing the transition code completely.